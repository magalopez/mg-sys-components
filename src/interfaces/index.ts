import { Props as ProductButtonsProps } from '../components/ProductCard/ProductButtons'
import { Props as ProductCardProps } from '../components/ProductCard/ProductCard';
import { Props as ProductImageProps } from '../components/ProductCard/ProductImage'
import { Props as ProductTitleProps } from '../components/ProductCard/ProductTitle'

export interface Product {
  id    : string;
  img?  : string;
  title : string;
}

export interface ProductContextProps {
  counter    : number;
  maxCount?  : number;
  product    : Product;
  increaseBy : (value: number) => void;
}

export interface ProductCardHOCProps {
  ({ children, product, className } : ProductCardProps) : JSX.Element,
  Buttons : (Props : ProductButtonsProps) => JSX.Element;
  Image   : (Props : ProductImageProps) => JSX.Element;
  Title   : (Props : ProductTitleProps) => JSX.Element;
}

export interface OnChangeArgs {
  count   : number;
  product : Product;
}

export interface ProductInCart extends Product {
  count : number
}

export interface ShoppingCartProps {
  [key: string] : ProductInCart
}

export interface InitialValuesProps {
  count?    : number;
  maxCount? : number;
}

export interface ProductCardHandlersArgs {
  count             : number;
  isMaxCountReached : boolean;
  maxCount?         : number;
  product           : Product;

  handleDecrement?  : () => void;
  handleIncrement?  : () => void;
  increaseBy        : (value: number) => void;
  reset             : () => void;
}