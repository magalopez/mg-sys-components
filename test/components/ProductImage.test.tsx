import React from 'react'
import renderer from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components';

describe('ProductImage', () => {
  test('should show Image', () => {
    const component = renderer.create(
      <ProductImage img="https://image.jpg" />
    )
    
    console.log(component.toJSON())

    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should show product object img ', () => {
    const product = {
      id: '1',
      title: 'Product Hellow',
      img: 'https://image-hellow.jpg'
    }

    const component = renderer.create(
      <ProductCard product={product}>
        {() => <ProductCard.Image/>}
      </ProductCard>
    )

    console.log(component.toJSON())
    expect(component.toJSON()).toMatchSnapshot()
  })
})