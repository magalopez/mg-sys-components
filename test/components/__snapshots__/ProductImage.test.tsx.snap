// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should show Image 1`] = `
<img
  alt="Product Image"
  className="productImg "
  src="https://image.jpg"
/>
`;

exports[`ProductImage should show product object img  1`] = `
<div
  className="productCard "
>
  <img
    alt="Product Image"
    className="productImg "
    src="https://image-hellow.jpg"
  />
</div>
`;
